add daisyui and light theme

hypy13 opened the pull request:
hey pal
thanks for your template
i have added light theme, check it out, and if you like it merge to your own project :)

djalig committed:
add daisyui and light theme

hypy13 committed:
Merge branch 'Tobi-De:main' into main

djalig committed:
exclude .idea

djalig committed:
Merge branch 'main' of https://github.com/hypy13/hy3

djalig committed:
exclude .idea

djalig committed:
update theme chooser, but there is issue if page loads by htmx

hypy13 commented:
heres demo:
https://hy3.ir/

Tobi-De commented:
> heres demo: https://hy3.ir/

Nice, I really like the background! Did you rewrite the whole thing to use Daisy?  
A small nitpick: something feels a bit off with the text—it blends too much with the background, in my opinion.

hypy13 commented:
hmm , maybe its because theme
i used cupcake for light theme and Halloween for dark theme

maybe switching it to normal light & dark make it better

Tobi-De commented:
btw the theme switcher is not working

> btw i recently find out https://nuejs.org/docs/

Yep, it's nice. I'm not really the target for this, though; it looks frontend-oriented, which I'm not.

hypy13 commented:
> btw the theme switcher is not working

its bcz of htmx , when it loads page by ajax, js events not working

but if you refresh the page it works again
how can we fix it?

hypy13 commented:
> Did you rewrite the whole thing to use Daisy?

not actually 
i used it to add themes  and tbh i rewrite some text colors classes

Tobi-De commented:
oh you already fixed the switcher issue, how ?

Tobi-De commented:
Oh I see, it works the first time, when you start navigating around it does not work anymore
feels like I had this issue before, don't remember how I fixed it, maybe reevaluate the theme script on htmx After request event or something

hypy13 commented:
i not familiar with htmx, 
i checked the nav links but couldnt figure out where it configured to load page by htmx

can we load just content block?
 {% block content %}
{% endblock content %}

not the whole html page?

Tobi-De commented:
> i not familiar with htmx, i checked the nav links but couldnt figure out where it configured to load page by htmx
> 
> can we load just content block? {% block content %} {% endblock content %}
> 
> not the whole html page?

Yes, or you can just remove HTMX, frankly it's not doing much here, if I'm not mistaken, the only change you need is here:  
https://github.com/Tobi-De/pw/blob/main/site/templates/base.html#L32  

Remove `hx-boost="true"` and `hx-ext="head-support"` (optional but unnecessary without `hx-boost`).  

By default, `hx-boost` will replace only the body, but on some pages (like blog post detail), some meta elements need to change. That's why I included the head-support extension to update the head as well.

djalig committed:
fix theme change by using htmx custom x-get / replace cupcake to light theme

hypy13 commented:
i switched cupcake to light 
and 
i read htmx documentations
i fixed it by these:

make the wrapper block like this:
<div id="main-content" hx-boost="true" hx-target="#content" hx-swap-oob="true">
    {% block content %}
    {% endblock content %}
</div>
hx-swap-oob="true" this attr is important part, which will replace new page content to here
---
then 
nav links like this:
<li>
                    <a  hx-get="/projects/" hx-target="#main-content" hx-push-url="true" href="/projects/"
                       :class="path.includes('projects') ? 'text-teal-400' : ''"
                       class="relative block px-3 py-2 transition hover:text-teal-400">Projects
                    </a>
                </li>                
 ---
